sips/sip-temporary_title.md

```diff
@@ -0,0 +1,50 @@
+| SIP-Number          | <Leave this blank; it will be assigned by a SIP Editor> |
+| ---:                | :--- |
+| Title               | Add more tx_context |
+| Description         | Add tx_context to get gas sender and transaction signer of the transaction |
+| Author              | Eason Chen (EasonC13) |
+| Editor              | <Leave this blank; it will be assigned by a SIP Editor> |
+| Type                | Standard |
+| Category            | Framework |
+| Created             | 2024-01-20 |
+| Comments-URI        | <Leave this blank; it will be assigned by a SIP Editor> |
+| Status              | <Leave this blank; it will be assigned by a SIP Editor> |
+| Requires            | <Optional; SIP number(s), comma separated> |
+
+
+## Abstract
+
+Can we get the gas sender of the transaction in Move contract?
+
+In Move, currently, we can get the transaction sender by tx_context::sender(ctx). 
+
+Curious if we can add the feature to get the gas sender, in my case is the sponsor of a transaction, by `tx_context::gas_sender(ctx)`
+
+Moreover, it would be great to check if the tx is signed by someone, maybe by `tx_context::is_signed_by(ctx, address)` and `tx_context::signers(ctx)`.
+
+## Motivation
+
+Regarding the use of tx_context::gas_sender(ctx), our service includes sponsored transactions to create objects. Therefore, we aim to delete objects that have been sponsored to create by us in order to collect storage rebates and prevent individuals from abusing sponsored transactions to create objects with sponsor and then deleting them using their own addresses to get rebate from sponsor.
+
+Regarding the use of tx_context::is_signed_by(ctx, address), we plan to run some transaction that need multiple signers to proceed.
+
+## Specification
+
+`tx_context::gas_sender(ctx)` return `address` of the gas sender.
+
+`tx_context::is_signed_by(ctx, address)` return `boolean`.
+
+`tx_context::signers(ctx)` return `vector<address>`.
```

Comment:
Couple of edge cases to think about here:
- What order does it return the addresses in? I would vote for gas sender last, but it doesn't matter as long as it's clear/consistent
- This should always consist of distinct addresses, right

Reply:
> What order does it return the addresses in?

At SDK, When we run `executeTransactionBlock` we pass the signature in as an array. Maybe the order can be the same as the array.

```
result = await suiClient.executeTransactionBlock({
        transactionBlock: sponsorSignedTxBlockBytes,
        signature: [sponsorSignature, clientSignature],
        options: {
          showEffects: true,
          showEvents: true,
          showObjectChanges: true,
        },
      });
```